docs/topics/extensions.md

```diff
@@ -0,0 +1,93 @@
+
+There are some changes to the protocol that extensions *do not* allow. These are:
+
+- Changing the definition of core data structures. Adding new fields or removing required fields to protocol-defined data structures is not supported. Extensions are expected to place custom attributes in the `metadata` map that is present on core data structures.
+- Adding new values to enum types. Instead, extensions should use existing enum values and annotate additional semantic meaning in the `metadata` field.
+
+These limitations exist to prevent extensions from breaking core type validations that clients and agents perform.
+
+### Architecture Overview
+
+[TODO: Diagram]
+
+- Extensions are defined by a specification document
+- The specification document defines the identifiers for the extension
+- The AgentCard declares support for an extension by referencing the identifier and providing any configuration
+- An implementation is built according to the specification
+- A client has support for various extensions
+- The client indicates extensions to activate in its request to the agent
+- Extensions can change the content of data that passes over the protocol
+
+## Extension Declaration
+
+Agents declare their support for extensions in their `AgentCard` by including `AgentExtension` objects in their `AgentCapabilities` object.
+
+- `uri`: The URI of the extension. This is an arbitrary identifier that the extension specification defines. Implementations of an extension use this URI to identify when to activate, and clients use this to determine extension compatibility.
+- `required`: Whether the agent requires clients to use this extension.
+- `description`: A description of how the agent uses the declared extension. Full details of a extension are intended to be in an extension specification. This field is useful to explain the connection between the agent and the extension.
+- `params`: Extension-specific configuration. The expected values to be placed in this field, if any, are defined by the extension specification. This field can be used for specifying parameters of the extension or declaring additional agent-specific data.
+
+### Required Extensions
+
+While extensions are a means of enabling additional functionality, we anticipate that some agents will have stricter requirements than those expressible by the core A2A protocol. For example, an agent may require that all incoming messages are cryptographically signed by their author. Extensions that are declared `required` are intended to support this use case.
+
+When an AgentCard declares a required extension, this is a signal to clients that some aspect of the extension impacts how requests are structured. Agents should not mark data-only extensions as required, since there is no direct impact on how requests are made to the agent.
```

Comment:
```suggestion
When an `AgentCard` declares a required extension, this is a signal to clients that some aspect of the extension impacts how requests are structured. Agents should not mark data-only extensions as required, since there is no direct impact on how requests are made to the agent.
```